docs/manual/spec.md

```diff
@@ -509,6 +509,9 @@ can just create the directory. It accepts a number of options:
 -n DIR      set the name of build directory (default is `%{name}-%{version}`)
 -T          skip the default unpacking of the first source (used with
             `-a` or `-b`)
+-p          ensures that the sources will be in the default build directory,
```

Comment:
How would this be exposed in `%autosetup`? We can't use `-p` there since that's for patchlevel passed to `patch`. Maybe instead use `-C` and describe it as `Create the build directory and ensure the archive contents are unpacked there, stripping the top level directory in the archive if it exists`.

Reply:
It isn't yet. By now it's only used in `%setup`, but we can add it in `%autosetup` too and pass it along the chain.
-C is fine for me, it reminds the `tar -C` command which is used in rpmuncompress